tasks/systemd.py

```diff
@@ -0,0 +1,106 @@
+"""
+Systemd test
+"""
+import contextlib
+import logging
+import re
+import time
+
+from cStringIO import StringIO
+from teuthology.orchestra import run
+
+log = logging.getLogger(__name__)
+
+
+@contextlib.contextmanager
+def task(ctx, config):
+    """
+      - tasks:
+          ceph-deploy:
+          systemd:
+
+    Test ceph systemd services can start, stop and restart and
+    check for any failed services and report back errors
+    """
+    for remote, roles in ctx.cluster.remotes.iteritems():
+        remote.run(args=['sudo', 'ps', '-eaf', run.Raw('|'),
+                         'grep', 'ceph'])
+        r = remote.run(args=['sudo', 'systemctl', 'list-units', run.Raw('|'),
+                             'grep', 'ceph'], stdout=StringIO(),
+                       check_status=False)
+        log.info(r.stdout.getvalue())
+        if r.stdout.getvalue().find('failed'):
+            log.info("Ceph services in failed state")
+
+        # test overall service stop and start
+        log.info("Stopping all Ceph services")
+        remote.run(args=['sudo', 'systemctl', 'stop', 'ceph.target'])
+        r = remote.run(args=['sudo', 'systemctl', 'status', 'ceph.target'],
```

Comment:
@ktdreyer the status works, the one that doesn't work is the ceph-disk service that comes up in failed state for which I am ignoring the errror for now until its fixed(line 28)
http://pulpito.ceph.com/vasu-2016-06-21_14:32:12-smoke:systemd-jewel---basic-vps/

2016-06-21T17:11:01.363 INFO:teuthology.orchestra.run.vpm097:Running: 'sudo systemctl list-units | grep ceph'
2016-06-21T17:11:01.452 INFO:tasks.systemd:Ceph services in failed state